### DISCUSSION TITLE
Alertmanager outgoing e-mails with custom subject

### DISCUSSION THREAD

**Discussion by ImperrattoR** [okd-project/okd#1215] [Q&A] [answered]
Hello

I want to figure out how to add text at the begining of subject of my alert mails send by alertmanager.
Is it posible?

Now my email subject looks like:
[FIRING:1]  (CoreDNSErrorsHigh openshift-monitoring/k8s warning)

I want to change it to:
OKD-text [FIRING:1]  (CoreDNSErrorsHigh openshift-monitoring/k8s warning)

**Comment by JaimeMagiera** [comment]
This KB is particular to slack messages, but it lays out the general premise. You modify the configuration file, adding the template in the respective section.

https://access.redhat.com/solutions/4142511

**Reply by ImperrattoR** [reply]
In alertmanager .yaml I have: 
headers:
  subject: '{{ template "email.subject" . }}'

All I want to know is how can I add text in front of email subject template.
Something like:
 subject: custom text '{{ template "email.subject" . }}'

**Comment by JaimeMagiera** [comment, marked as answer]
You have a couple options. Putting the extra words after the quote but before the jinja block should work. 

'Foobar {{ template "email.subject" . }}'

For more granularity, you could override the template. The message below explains well, even if you don't want a dynamic subject. 

https://www.mail-archive.com/prometheus-users@googlegroups.com/msg09181.html

Here is the Prometheus documentation. 

https://prometheus.io/docs/alerting/latest/notification_examples/

**Comment by ImperrattoR** [comment]
Thanks for answear.
I have one more question.
Is there a chance to configure subject like:
subject: '[okd][severity type label] {{ template "email.default.subject" . }}'

Now I have subject configured like
subject: '[OKD-TEST] {{ template "email.default.subject" . }}'
I tried something like this:
subject: '[OKD-TEST] {{ $labels.severity }} {{ template "email.default.subject" . }}'
But it doesn't work.

I want output like this:
[okd] [warning][RESOLVED] openshift-monitoring (email openshift-monitoring/k8s alertmanager-main warning)

**Reply by JaimeMagiera** [reply]
Sorry, I don't know. At this point, your questions are more appropriate for the Prometheus community.
